### PR TITLE
Add v1.0.0 release notes

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*
- Add v1.0.0 release notes

TODO:

- [x] Add Twitter, YouTube, Discord links
- [x] Add final numbers for commit count
- [x] Add summary and intro @Innixma 
- [x] Add Contributor Thanks @Innixma 
- [x] Add Tabular notes @Innixma 
- [x] Add Tabular Benchmark results @Innixma 
- [x] Add AutoMM notes @zhiqiangdon @FANGAreNotGnu @taoyang1122 
- [x] Add TimeSeries notes @shchur
- [x] Add TimeSeries paper comment @shchur 
- [x] Add EDA status update @Innixma
- [x] Add TabRepo paper comment @Innixma
- [ ] Add video link @Innixma

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/whats_new/v1.0.0.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+* Upgraded torch to 2.0. @zhiqiangdon (#3404)
+* Upgraded lightning to 2.0 @zhiqiangdon (#3419)
+* Upgraded torchmetrics to 1.0 @zhiqiangdon (#3422)
+
+### Deprecations
+
+* Deprecated the `FewShotSVMPredictor` in favor of the new `few_shot_classification` problem type. @zhiqiangdon (#3699)
+* Deprecated the `AutoMMPredictor` in favor of `MultiModalPredictor`. @zhiqiangdon (#3650)
+* Deprecated the `config` argument in the fit API. @zhiqiangdon (#3679)
+* Deprecated the `init_scratch` and `pipeline` arguments in init API. @zhiqiangdon (#3668)
```
Comment (shchur) [inline docs/whats_new/v1.0.0.md:80]: Should we maybe move these to the `Deprecations` section below for consistency?

File: docs/whats_new/v1.0.0.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+we are excited to share with you the culmination of our efforts to create and democratize the most powerful, easy to use, and feature rich automated machine learning system in the world.
+AutoGluon 1.0 comes with transformative enhancements to predictive quality resulting from the combination of multiple novel ensembling innovations, spotlighted below.
+Besides performance enhancements, many other improvements have been made that are detailed in the individual module sections.
+
+Note: Loading models trained on older versions of AutoGluon is not supported. Please re-train models using AutoGluon 1.0.
+
+This release contains XXX commits from 17 contributors!
+
+Special thanks to @LennartPurucker for leading development of dynamic stacking, @geoalgo for co-authoring [TabRepo](https://github.com/autogluon/tabrepo) to enable Zeroshot-HPO, @ddelange for helping to add Python 3.11 support, and @mlgowacki100 for providing numerous comments and suggestions
```
Comment (gidler) [inline docs/whats_new/v1.0.0.md:14]: I would put a period at the end of this sentence.

File: docs/whats_new/v1.0.0.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+We are excited to see what our users can accomplish with AutoGluon 1.0's enhanced performance.
+As always, we will continue to refine and enhance AutoGluon in future releases to push the boundaries of AutoML forward for all.
+
+## Spotlight: Newly Published Papers
+
+### AutoGluon-TimeSeries: AutoML for Probabilistic Time Series Forecasting
+
+We have published a [paper on AutoGluon-TimeSeries](https://openreview.net/forum?id=XHIY3cQ8Tew) at AutoML Conference 2023 ([YouTube Video](https://www.youtube.com/watch?v=niLmfjXeHnE)).
+In the paper, we benchmarked AutoGluon and popular open-source forecasting frameworks (including DeepAR, TFT, AutoARIMA, AutoETS, AutoPyTorch).
+AutoGluon produces SOTA results in point & probabilistic forecasting, and even **achieves 65% win rate against the best-in-hindsight combination of models**.
```
Comment (gidler) [inline docs/whats_new/v1.0.0.md:88]: Not sure we need a `&` here when we have stuck with `and` everyone else in the notes.

File: docs/whats_new/v1.0.0.md [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+- Add reference to the publication on AutoGluon-TimeSeries to README @shchur (#3482)
+- Align API of `TimeSeriesPredictor` with `TabularPredictor`, remove deprecated methods @shchur (#3714, #3655, #3396)
+- General bug fixes and improvements @shchur(#3758, #3756, #3755, #3754, #3746, #3743, #3727, #3698, #3654, #3653, #3648, #3628, #3588, #3560, #3558, #3536, #3533, #3523, #3522, #3476, #3463)
+
+## EDA
+
+The EDA module will be released at a later time, as it requires additional development effort before it is ready for 1.0.
+We will make an announcement when EDA is ready for release. For now, please continue to use `"autogluon.eda==0.8.2"`.
+
+## Deprecations
```
Comment (gidler) [inline docs/whats_new/v1.0.0.md:253]: We use periods intermittently in this section. We should make it more consistent.

**Review by gidler — APPROVED** [review]
Added a few punctuation comments.

File: docs/whats_new/v1.0.0.md (hunk 2 above)
Comment (Innixma) [reply docs/whats_new/v1.0.0.md:14]: added

File: docs/whats_new/v1.0.0.md (hunk 3 above)
Comment (Innixma) [reply docs/whats_new/v1.0.0.md:88]: added

File: docs/whats_new/v1.0.0.md (hunk 4 above)
Comment (Innixma) [reply docs/whats_new/v1.0.0.md:253]: updated

File: docs/whats_new/v1.0.0.md [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+AutoGluon 1.0 has a 95%+ win-rate against traditional tabular models, including **a 99% win-rate vs LightGBM and a 100% win-rate vs XGBoost**.
+AutoGluon 1.0 has between an 82% and 94% win-rate against other AutoML systems.
+For all methods, AutoGluon is able to achieve >10% average loss improvement (Ex: Going from 90% accuracy to 91% accuracy is a 10% loss improvement).
+**AutoGluon 1.0 achieves first place in 63% of tasks**, with lightautoml having the second most at 12% (AutoGluon 0.8 previously took first place 48% of the time).
+AutoGluon 1.0 even achieves a 7.4% average loss improvement over AutoGluon 0.8!
+
+| Method                       | AG Winrate | AG Loss Improvement | Rescaled Loss |     Rank | Champion |
+|:-----------------------------|:-----------|:--------------------|--------------:|---------:|:---------|
+| AutoGluon 1.0 (Best, 4h8c)   | **-**      | **-**               |      **0.04** | **1.95** | **63%**  |
+| lightautoml (2023, 4h8c)     | 84%        | 12.0%               |           0.2 |     4.78 | 12%      |
```
Comment (tonyhoo) [inline docs/whats_new/v1.0.0.md:50]: Can we include the version information for other libraries here, similar to what we do for AG, to maintain consistency?

File: docs/whats_new/v1.0.0.md [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+AutoGluon produces SOTA results in point and probabilistic forecasting, and even **achieves 65% win rate against the best-in-hindsight combination of models**.
+
+### TabRepo: A Large Scale Repository of Tabular Model Evaluations and its AutoML Applications
+
+We have published a paper on Tabular Zeroshot-HPO ensembling simulation to arXiv ([Paper Link](https://arxiv.org/pdf/2311.02971.pdf), [GitHub](https://github.com/autogluon/tabrepo)).
+This paper is key to achieving the performance improvements seen in AutoGluon 1.0, and we plan to continue to develop the code-base to support future enhancements.
+
+### XTab: Cross-table Pretraining for Tabular Transformers
+
+We have published a paper on tabular Transformer pre-training at ICML 2023 ([Paper Link](https://arxiv.org/abs/2305.06090)).
```
Comment (tonyhoo) [inline docs/whats_new/v1.0.0.md:100]: Can we add github link here as well? Potentially this one: https://github.com/BingzhaoZhu/XTab

File: docs/whats_new/v1.0.0.md [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+@shchur, @zhiqiangdon, @Innixma, @prateekdesai04, @FANGAreNotGnu, @yinweisu, @taoyang1122, @LennartPurucker, @Harry-zzh, @AnirudhDagar, @jaheba, @gradientsky, @melopeo, @ddelange, @tonyhoo, @canerturkmen, @suzhoum
+
+## Spotlight: Tabular Performance Enhancements
+
+AutoGluon 1.0 features **major enhancements to predictive quality**, establishing a new state-of-the-art in Tabular modeling.
+To the best of our knowledge, AutoGluon 1.0 marks the largest leap forward in the state-of-the-art for tabular data since the [original AutoGluon paper](https://arxiv.org/abs/2003.06505) from March 2020.
+The enhancements come primarily from two features:
+Dynamic stacking to [mitigate stacked overfitting](https://github.com/autogluon/autogluon/issues/2779#issuecomment-1736468165),
+and a new learned model hyperparameters portfolio via Zeroshot-HPO, obtained from the newly released [TabRepo](https://github.com/autogluon/tabrepo) ensemble simulation library.
+Together, they lead to a **75% win-rate compared to 0.8 with faster inference speed, lower disk usage, and higher stability.**
```
Comment (tonyhoo) [inline docs/whats_new/v1.0.0.md:30]: [minor] compared to AutoGluon version 0.8

File: docs/whats_new/v1.0.0.md [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+    * Using [SAM-ViT Huge](https://huggingface.co/facebook/sam-vit-huge) by default (GPU memory > 25GB required).
+* Few Shot Classification
+    * Added the new `few_shot_classification` problem type for training few shot classifiers on images or texts. @zhiqiangdon (#3662, #3681, #3695)
+    * Leveraging image/text foundation models to extract features and train SVM classifiers.
+    * Added one [few shot classification tutorial](https://auto.gluon.ai/stable/tutorials/multimodal/advanced_topics/few_shot_learning.html). @zhiqiangdon (#3662)
+* Supported [torch.compile](https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html) for faster training (experimental and torch >=2.2 required) @zhiqiangdon (#3520).
+
+### Performance Improvements
+
+* Improved default image backbones, achieving a 100% win-rate on the image benchmark. @taoyang1122 (#3738)
```
Comment (tonyhoo) [inline docs/whats_new/v1.0.0.md:181]: @taoyang1122, could you please provide the benchmark config link here? Additionally, please specify the version (assuming version 0.8.2) that we have compared against.

Comment (taoyang1122) [reply docs/whats_new/v1.0.0.md:181]: Yes, it is in comparison with 0.8.2. The config is just the default config.

File: docs/whats_new/v1.0.0.md (hunk 6 above)
Comment (Innixma) [reply docs/whats_new/v1.0.0.md]: added

File: docs/whats_new/v1.0.0.md (hunk 5 above)
Comment (Innixma) [reply docs/whats_new/v1.0.0.md:55]: This is noted in the AMLB paper

**Pull request merged**